Assets/_BForBoss/_Weapons/Scripts/BulletSpawner.cs

```diff
@@ -0,0 +1,52 @@
+using Perigon.Utility;
+using Sirenix.OdinInspector;
+using UnityEngine;
+
+namespace Perigon.Weapons
+{
+
+    public enum BulletTypes
+    {
+        RifleNoPhysics,
+        RiflePhysics
+    }
+    
+    public class BulletSpawner : MonoBehaviour
+    {
+        [SerializeField] private BulletBehaviour[] _bulletPrefabs;
+        [InlineEditor]
+        [SerializeField] private BulletPropertiesScriptableObject _bulletPropertiesScriptableObject;
+        // placeholder for more robust bullet switching
+        [SerializeField] private bool _usePhysics = false;
```

Comment:
I think to simplify things, we can remove this. 

And then the designer can just pass in the correct `BulletBehaviour` Prefab of their choice.  Instead of having an array of `BulletBehaviour`

The Physics bullet behaviour is currently out of scope. But it's a nice to keep in mind for the future.

Reply:
I was kind of thinking we could change `BulletSpawner.SpawnBullet` to take a `BulletType` parameter, and then give a `BulletType` property to the `WeaponScriptableObjects` - do you that would work, or is that overengineering it a bit?